src/index.ts

```diff
@@ -31,9 +40,10 @@ export default class DataSource {
 
     if (opts['headers']) {
       params['mi_include_headers'] = true;
+      opts['headers'] = {};
```

Comment:
Ah I see why this is necessary. Wondering if we should rethink the API for new versions moving forward to be a bit clearer about how you interact with `sorcerer` in general...

`getAllRows` was meant to be a "super easy" way to not have to think about making requests and getting rows back. This change would make it so that just including a header override would make it so _any_ requests to `getAllRows` will always return headers merged in. I know that we also talked about how most users almost _always_ want the headers merged in already.

This change is fine, maybe we can get together in the future about what `data-source.js` could ideally look like? Cause I also want to eventually include `data-source.js` in other places too.

Reply:
Yea I struggled with this a little as ideally it should be something other than `headers` so we could use that when we make the request, which is why I emptied it out. 

I was actually thinking this maybe something we can configure in the DS UI (the cachetime) as all of the other headers should be set up in there.